app/crud/async_db/vector_tiles/filters.py

```diff
@@ -0,0 +1,77 @@
+from typing import Any, Dict, List, Optional, Tuple, Union
+
+from fastapi import HTTPException
+from shapely.geometry import box
+from sqlalchemy.sql.elements import TextClause
+
+from ....application import db
+from ....utils.geostore import get_geostore_geometry
+
+Filter = Tuple[TextClause, Dict[str, Any]]
+Bounds = Tuple[float, float, float, float]
+
+
+async def geometry_filter(
+    geostore_id: Optional[str], bounds: Bounds
```

Comment:
Why is geostore_id optional? Is it supposed to do something if it's None (like just return the whole bounding box)?

Reply:
All the filters (except date) are optional.
In the route I simply forward the user input (which is either `None` or a value) to the filter. Only here i check if the value is empty and if yes return and and convert it into a Text Clause. Only here I check the value and return `None` or a TextClause.

In the route I then only filter out empty filters.

https://github.com/wri/gfw-tile-cache/blob/feature/gino/app/routes/nasa_viirs_fire_alerts/vector_tiles.py#L79-L86

Filter out `None` values in the route before converting them to TextClause seem to verbose to me.